transformations/chinese_simplified_traditional_perturbation/transformation.py

```diff
@@ -1,57 +1,82 @@
 import itertools
 import random
-import spacy
-import numpy
+import opencc
+import json
+import os
 
 from interfaces.SentenceOperation import SentenceOperation
 from tasks.TaskTypes import TaskType
-from SoundsLike.SoundsLike import Search
-from spacy.attrs import LOWER, POS, ENT_TYPE, IS_ALPHA
-from spacy.tokens import Doc
+import shutil
+import jieba
+import pypinyin
 
-def close_homophones_swap(text, corrupt_prob, seed=0, max_outputs=1, nlp = None):
+
+def chinese_butter_finger(text,
+                          prob,
+                          seed,
+                          max_outputs,
+                          config
+                          ):
     random.seed(seed)
-    doc = nlp(text)
+
     perturbed_texts = []
-    spaces = [True if tok.whitespace_ else False for tok in doc]
-    for _ in range(max_outputs):
-        perturbed_text = []
-        for index, token in enumerate(doc):
-            if random.uniform(0, 1) < corrupt_prob:
-                try:
-                    replacement = random.choice(Search.closeHomophones(token.text))
-                    if replacement.lower()!=token.text.lower() and token.text.lower()!='a':
-                        perturbed_text.append(replacement)
-                    else:
-                        perturbed_text.append(token.text)
-                except:
-                    perturbed_text.append(token.text)
+
+    if(config == 's2t'):
+        converter = opencc.OpenCC('s2t.json')
+    elif (config == 't2s'):
+        converter = opencc.OpenCC('t2s.json')
+    else:
+        print("Configuration not specified")
+    for _ in itertools.repeat(None, max_outputs):
+        butter_text = ""
+        for chinese_character in text:
+            if random.random() <= prob:
+
+                new_chinese_character = converter.convert(chinese_character)
             else:
-                perturbed_text.append(token.text)
-        
-        textbf = []             
-        for index, token in enumerate(perturbed_text):
-            textbf.append(token)
-            if spaces[index]:
-                textbf.append(' ')
-        perturbed_texts.append(''.join(textbf))
+                new_chinese_character = chinese_character
+
+            butter_text += new_chinese_character
+        perturbed_texts.append(butter_text)
     return perturbed_texts
-       
-class CloseHomophonesSwap(SentenceOperation):
+
+
+
+
+"""
+Chinese Words and Characters Butter Fingers Perturbation
+"""
+
+class ChineseSimplifiedTraditionalPerturbation(SentenceOperation):
     tasks = [
         TaskType.TEXT_CLASSIFICATION,
-        TaskType.TEXT_TO_TEXT_GENERATION,
-        TaskType.TEXT_TAGGING,
+        TaskType.TEXT_TO_TEXT_GENERATION
     ]
-    languages = ["en"]
+    languages = ["zh"]
 
-    def __init__(self, seed=0, max_outputs=1):
-        super().__init__(seed)
+    def __init__(self, seed=0, max_outputs=1, prob=1, config = 's2t'):
```

Comment:
the code can use a bit of documentation on `prob`, `config`, etc.?

Reply:
Should the probability be 1 by default?
Maybe the wording of the variable can just be changed to address @tongshuangwu comment. Like changing `prob` to `character_replacement_prob` or similar and changing `config` with `converter_config` (i am not pleased with this either but you can find a better naming if you want to change it :) )

Reply:
I kept it as `prob` to be consistent with other transformations but I agree that it can be confusing! I'll rename the both of them and add documentation in the code as well as README.md

I'll call them `transformation_prob` and `converter_config`.

I put the probability as 1 by default since most of the time you want to convert the whole text that is in Simplified to Traditional or vice-versa. This is because when you normally type using a Pinyin keyboard - you would be using either Simplified or Traditional. Typically, you won't switch between the two when you type using a Pinyin keyboard. There will be words/characters that are the same in both Simplified and Traditional though. :)